GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using UnityEngine;
+
+namespace Turn_based_assets.Scripts
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
+        [SerializeField] private GameObject cube;
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        private Plane _plane;
+        private Vector3 distanceFromCamera;
```

Comment:
You have public fields here. Please rectify.

Also, nonpublic field convention appears to be conflicting here. Either pick `_camelCase` or `camelCase` for your project but do not use both as it causes confusion.